diff --git a/libs/core/threading_base/include/hpx/threading_base/annotated_function.hpp b/libs/core/threading_base/include/hpx/threading_base/annotated_function.hpp
index 10949b9f4c2..75df97c954d 100644
--- a/libs/core/threading_base/include/hpx/threading_base/annotated_function.hpp
+++ b/libs/core/threading_base/include/hpx/threading_base/annotated_function.hpp
@@ -33,7 +33,7 @@ namespace hpx { namespace util {
 #if defined(HPX_HAVE_THREAD_DESCRIPTION)
     ///////////////////////////////////////////////////////////////////////////
 #if defined(HPX_COMPUTE_DEVICE_CODE)
-    struct HPX_NODISCARD annotate_function
+    struct annotate_function
     {
         HPX_NON_COPYABLE(annotate_function);
 
@@ -47,7 +47,7 @@ namespace hpx { namespace util {
         HPX_HOST_DEVICE ~annotate_function() {}
     };
 #elif HPX_HAVE_ITTNOTIFY != 0
-    struct HPX_NODISCARD annotate_function
+    struct annotate_function
     {
         HPX_NON_COPYABLE(annotate_function);
 
@@ -68,7 +68,7 @@ namespace hpx { namespace util {
         hpx::util::itt::task task_;
     };
 #else
-    struct HPX_NODISCARD annotate_function
+    struct annotate_function
     {
         HPX_NON_COPYABLE(annotate_function);
 
@@ -210,7 +210,7 @@ namespace hpx { namespace util {
 
 #else
     ///////////////////////////////////////////////////////////////////////////
-    struct HPX_NODISCARD annotate_function
+    struct annotate_function
     {
         HPX_NON_COPYABLE(annotate_function);
 
