--- src/otf2_archive_int.c-org	2021-09-06 09:01:37.673187730 +0200
+++ src/otf2_archive_int.c	2021-09-06 09:02:22.376037402 +0200
@@ -1083,7 +1083,7 @@
     archive->global_comm_context  = globalCommContext;
     archive->local_comm_context   = localCommContext;
 
-    OTF2_ErrorCode status;
+    OTF2_ErrorCode status = OTF2_SUCCESS;
 
     /* It is time to create the directories by the root rank. */
     if ( archive->file_mode == OTF2_FILEMODE_WRITE )
